credentials/credentials_test.go

```diff
@@ -147,6 +147,7 @@ func launchServer(t *testing.T, hs serverHandshake, done chan AuthInfo) net.List
 // Is run in a seperate goroutine.
 func serverHandle(t *testing.T, hs serverHandshake, done chan AuthInfo, lis net.Listener) {
 	serverRawConn, err := lis.Accept()
+	defer serverRawConn.Close()
```

Comment:
two problems here:
i) Accept may return a nil serverRawConn when err is not nil;
ii) it may close the connection too quick so that the client has not read the response.